aquarius/events/events_monitor.py

```diff
@@ -181,51 +182,44 @@ def process_block_range(self, from_block, to_block):
             self._chain_id,
         ]
 
-        for event in self.get_event_logs(EVENT_METADATA_CREATED, from_block, to_block):
-            dt_contract = self._web3.eth.contract(
-                abi=ERC721Template.abi, address=event.address
-            )
-            receipt = self._web3.eth.get_transaction_receipt(
-                event.transactionHash.hex()
-            )
-            event_object = dt_contract.events.MetadataCreated().processReceipt(receipt)[
-                0
-            ]
-            try:
-                event_processor = MetadataCreatedProcessor(
-                    *([event_object, dt_contract, receipt["from"]] + processor_args)
-                )
-                event_processor.process()
-            except Exception as e:
-                logger.exception(
-                    f"Error processing create metadata event: {e}\n" f"event={event}"
-                )
+        event_processors = {
+            "EVENT_METADATA_CREATED": MetadataCreatedProcessor,
+            "EVENT_METADATA_UPDATED": MetadataUpdatedProcessor,
+            "EVENT_METADATA_STATE": MetadataStateProcessor,
+        }
 
-        for event in self.get_event_logs(EVENT_METADATA_UPDATED, from_block, to_block):
-            dt_contract = self._web3.eth.contract(
-                abi=ERC721Template.abi, address=event.address
-            )
-            receipt = self._web3.eth.get_transaction_receipt(
-                event.transactionHash.hex()
-            )
-            event_object = dt_contract.events.MetadataUpdated().processReceipt(receipt)[
-                0
-            ]
-            try:
-                event_processor = MetadataUpdatedProcessor(
-                    *([event_object, dt_contract, receipt["from"]] + processor_args)
+        for event_name in event_processors:
+            for event in self.get_event_logs(
```

Comment:
I'd rather move the contents of this into a separate function. Like we have handle_order_started, we can have handle_regular_event_processing(). The complexity won't change, but the readability will improve